modular_splurt/code/game/object/items/lewd_items/rope.dm

```diff
@@ -0,0 +1,482 @@
+#define ROPE_STATE_UNTIED 0
+#define ROPE_STATE_DECIDING_OBJECT 1
+#define ROPE_STATE_TIED 2
+
+#define ROPE_TARGET_LEGS 0
+#define ROPE_TARGET_HANDS_IN_FRONT 1
+#define ROPE_TARGET_HANDS_BEHIND 2
+#define ROPE_TARGET_LEGS_OBJECT 3
+#define ROPE_TARGET_HANDS_OBJECT 4
+
+#define ROPE_OBJECT_IMMOVABLE 0
+#define ROPE_OBJECT_MOVABLE 1
+
+#define ROPE_MAX_DISTANCE_OBJECT 1
+#define ROPE_MAX_DISTANCE_MASTER 2
+#define ROPE_MAX_DISTANCE_SNAP 4
+#define ROPE_OBJECT_IMMOVABLE_SLOWDOWN 32
+#define ROPE_SELF_APPLY_INSTANT 0
```

Comment:
These should be declared in code/__SPLURTCODE/DEFINES if they're necessary for other files, or undefined at the bottom of the file if they're not